brainfuck/bf.cpp

```diff
@@ -76,11 +75,11 @@ class Program {
 
 private:
 
-  Ops parse(string::iterator *iterator, string::iterator end) {
+  Ops parse(string::const_iterator &iterator, string::const_iterator end) {
     Ops res;
-    while (*iterator != end) {
-      char c = **iterator;
-      *iterator += 1;
+    while (iterator != end) {
+      const char c = *iterator;
```

Comment:
I guess we can optimize a little bit more by incrementing right here:
```
const auto c = *iterator++;
```

Reply:
Done